go-controller/pkg/node/gateway_iptables.go

```diff
@@ -330,8 +330,14 @@ func getGatewayIPTRules(service *kapi.Service, gatewayIPs []string) []iptRule {
 					rules = append(rules, getNodePortIPTRules(svcPort, clusterIP, svcPort.Port)...)
 				}
 			} else {
-				for _, gatewayIP := range gatewayIPs {
-					rules = append(rules, getNodePortIPTRules(svcPort, gatewayIP, svcPort.Port)...)
+				if dnatToPod {
+					for _, gatewayIP := range gatewayIPs {
+						rules = append(rules, getNodePortIPTRules(svcPort, gatewayIP, int32(svcPort.TargetPort.IntValue()))...)
+					}
+				} else {
+					for _, gatewayIP := range gatewayIPs {
+						rules = append(rules, getNodePortIPTRules(svcPort, gatewayIP, svcPort.Port)...)
```

Comment:
So I think this is a bug... The [SyncServices function for LGW](https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_localnet.go#L263) is the only place we call it... And `<NodeIP>:<svcIP>` Shouldn't ever be a real endpoint... Also in LGW the IPtables rules should always be DNATing straight to the ClusterIP of the service rather than the IP of the node itself. 

I'm going to leave this comment un-resolved as a note to self to fix this funtion